src/draw.js

```diff
@@ -217,52 +217,56 @@ export default class Draw extends API {
   }
 
   _setStyles() {
-    // drawn features style
-    this._map.addSource('draw', {
-      data: {
-        type: 'FeatureCollection',
-        features: []
-      },
-      type: 'geojson'
-    });
+    var self = this;
```

Comment:
I think using [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) as callbacks will bind `this` to the functions so that you don't have to alias `this` as `self`